remoteworkitem/flatten.go

```diff
@@ -0,0 +1,56 @@
+package remoteworkitem
+
+import (
+	"reflect"
+	"strings"
+)
+
+type stack []string
+
+func (s *stack) push(v string) {
+	*s = append(*s, v)
+}
+
+func (s *stack) pop() string {
+	l := len(*s)
+	if l == 0 {
+		return ""
+	}
+	stackReference := *s // because *s[:l-1] did not work.
+	poppedData := stackReference[l-1]
+	*s = stackReference[:l-1]
+	return poppedData
+}
+
+func (s stack) String() string {
+	return strings.Join(s, ".")
+}
+
+// MapFlattener defines the data structures and the implemetation of how to flatten a nested map.
+type MapFlattener struct {
```

Comment:
You don't need it as a global variable to make it a function. Use Flatten() as an API exposed function, then call a recursive helper function internally and pass the needed values as arguments.